pkg/descheduler/strategies/node_affinity.go

```diff
@@ -43,11 +43,18 @@ func RemovePodsViolatingNodeAffinity(ctx context.Context, client clientset.Inter
 			for _, node := range nodes {
 				klog.V(1).Infof("Processing node: %#v\n", node.Name)
 
-				pods, err := podutil.ListPodsOnANode(ctx, client, node, func(pod *v1.Pod) bool {
-					return podEvictor.IsEvictable(pod) &&
-						!nodeutil.PodFitsCurrentNode(pod, node) &&
-						nodeutil.PodFitsAnyNode(pod, nodes)
-				})
+				pods, err := podutil.ListPodsOnANode(
+					ctx,
+					client,
+					node,
+					podutil.WithFilter(func(pod *v1.Pod) bool {
+						return podEvictor.IsEvictable(pod) &&
+							!nodeutil.PodFitsCurrentNode(pod, node) &&
+							nodeutil.PodFitsAnyNode(pod, nodes)
+					}),
+					podutil.WithNamespaces(strategy.Params.Namespaces.Include),
```

Comment:
Ah, I see now how the strategies can easily control whether they implement the include/exclude. Cool!